ui/transactions_page.go

```diff
@@ -260,11 +260,14 @@ func (pg *transactionsPage) sortTransactions(common *pageCommon) {
 	}
 }
 
-func (pg *transactionsPage) goToTxnDetails(events []gesture.ClickEvent, changePage func(string), txn *wallet.Transaction) {
+func (pg *transactionsPage) goToTxnDetails(events []gesture.ClickEvent, common *pageCommon, txn *wallet.Transaction) {
 	for _, e := range events {
 		if e.Type == gesture.TypeClick {
 			*pg.walletTransaction = txn
-			changePage(PageTransactionDetails)
+
+			*common.returnPage = PageTransactions
+			common.setReturnPage(PageTransactions)
```

Comment:
This looks like the returnPage field is being set twice. It is set directly on line 268 once and a second time with a method 269.